sdk/core/src/main/java/com/klaviyo/core/config/FileLogger.kt

```diff
@@ -0,0 +1,647 @@
+    private fun flushBuffer() {
+        try {
+            synchronized(buffer) {
+                if (buffer.isEmpty()) return
+
+                val file = getCurrentFile()
+
+                // Check if we need to rotate files
+                // Calculate actual byte size (not character count) for UTF-8 compatibility
+                val bufferBytes = buffer.toString().toByteArray(Charsets.UTF_8).size
+                if (file.length() + bufferBytes >= maxFileSize) {
+                    rotateFiles()
+                }
+
+                // Append buffer to file (get again in case it was rotated)
+                getCurrentFile().appendText(buffer.toString())
+                buffer.clear()
+            }
+        } catch (e: Exception) {
+            // Log errors to logcat only, to avoid recursive errors
+            Log.Level.Error.log(TAG, "Failed to flush buffer", e)
+        }
+    }
+
+    /**
+     * Initialize the log file on first access.
+     * Reuses the most recent existing file if it's under maxFileSize,
+     * otherwise creates a new file.
+     */
+    @WorkerThread
+    private fun initializeLogFile(): File {
+        try {
+            val existingFiles = getLogFiles()
+            val mostRecentFile = existingFiles.firstOrNull()
+
+            // If we have a recent file that's not full, reuse it with a session marker
+            if (mostRecentFile != null && mostRecentFile.length() < maxFileSize) {
+                mostRecentFile.appendText("\n$SESSION_MARKER\n")
+                return mostRecentFile
+            }
+        } catch (e: Exception) {
+            Log.Level.Error.log(TAG, "Failed to check for existing log file, creating new", e)
+        }
+
+        // Otherwise, create a new file
+        return createNewLogFile()
+    }
+
+    /**
+     * Create a new log file with timestamp-based name
+     */
+    @WorkerThread
+    private fun createNewLogFile(): File {
+        val timestamp = fileNameFormat.format(Date())
+        return File(logDirectory, "$FILE_PREFIX$timestamp.$LOG_FILE_EXTENSION")
+    }
+
+    /**
+     * Get the current log file, should be used from background thread if possible
+     */
+    @WorkerThread
+    private fun getCurrentFile(): File {
+        return synchronized(buffer) {
+            currentFile ?: initializeLogFile().also { currentFile = it }
+        }
+    }
+
+    /**
+     * Rotate log files: delete oldest if over limit, create new file
+     */
+    @WorkerThread
+    private fun rotateFiles() {
+        try {
+            val files = getLogFiles()
+
+            // Delete oldest files if we exceed maxFiles
+            if (files.size >= maxFiles) {
+                files.drop(maxFiles - 1).forEach { it.delete() }
+            }
+
+            // Create new log file
+            synchronized(buffer) {
+                currentFile = createNewLogFile()
+            }
+        } catch (e: Exception) {
+            Log.Level.Error.log(TAG, "Failed to rotate files", e)
```

Comment:
Will these logs funnel back through here somehow?

Reply:
No, this is an alias to logcat directly, to avoid recursive calls